include/rocksdb/options.h

```diff
@@ -81,6 +81,11 @@ enum CompressionType : unsigned char {
   // eventually remove the option from the public API.
   kZSTDNotFinalCompression = 0x40,
 
+  // Fake compressors that simulate different errors. Used for test only.
+  kTestCompressionFail = 0x50,
+  kTestDecompressionFail = 0x51,
+  kTestDecompressionCorruption = 0x52,
```

Comment:
This is a public API that is exposed to users. Can we try to avoid adding things here just for test purpose?

Did you try to use sync point to alter compression outcome?

Reply:
Fixed using `TEST_SYNC_POINT_CALLBACK` instead.